src/Cache/RedisDistributedCache.cs

```diff
@@ -71,15 +93,12 @@ public T Get<T>(string key)
 
             using (var client = this.redisManager.GetClient())
             {
-                if (client.ContainsKey(key))
-                {
-                    var entry = client.Get<RedisDistributedCacheEntry<T>>(key);
-
-                    if (entry != null)
-                    {
-                        result = entry.Value;
-                    }
+                var typedClient = client.As<T>();
+                var redisHash = typedClient.GetHash<string>(key);
```

Comment:
should call this after the if (typedClient.ContainsKey(key)) ?

Reply:
Yes. Will move it to deeper scope